tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -134,18 +135,21 @@ public void CloseSpan(Span span)
             {
                 Profiler.Instance.ContextTracker.SetEndpoint(span.RootSpanId, span.ResourceName);
 
-                if (Iast.Iast.Instance.Settings.Enabled)
+                var iastInstance = Iast.Iast.Instance;
+                if (iastInstance.Settings.Enabled)
                 {
                     if (_iastRequestContext is { } iastRequestContext)
                     {
                         iastRequestContext.AddIastVulnerabilitiesToSpan(span);
-                        OverheadController.Instance.ReleaseRequest();
+                        iastInstance.OverheadController.ReleaseRequest();
                     }
                     else
                     {
                         IastRequestContext.AddIastDisabledFlagToSpan(span);
                     }
                 }
+
+                Security.Instance.ApiSecurity.ReleaseRequest();
```

Comment:
This makes the assumption that this span has also been acquired by `ApiSecurity`? That seems _likely_ but it's not guaranteed AFAICT. The IAST case is safer, because it's checking for the `iastRequestContext` first, but there's no such check here 🤔 Just flagging because it could mess with your sampling...

Reply:
it's a good point, we dont have a `securityApiContext` and as the HttpContext is gone by that stage, not sure where to look at, apart from creating a context for us as well 🤔